lib/ecto/changeset.ex

```diff
@@ -1217,6 +1217,16 @@ defmodule Ecto.Changeset do
     raise ArgumentError, "field names given to change/put_change must be atoms, got: `#{inspect(key)}`"
   end
 
+  defp put_change(data, changes, errors, valid?, key, new_value, {:parameterized, type, params}) do
+    old_value = Map.get(data, key)
+
+    case type.change(old_value, new_value, params) do
+      {:ok, change} -> {Map.put(changes, key, change), errors, valid?}
+      {:error, error} -> {changes, [{key, error} | errors], false}
```

Comment:
Please do the Keyword.pop bit in the original implementation too. :)

Reply:
I'm always trying to do 1 reasonable thing (with passing tests) for one commit and well describe what I do in commit message.

Updated as in previous comment even before saw this one. :-)